File: client/src/types/JsonModels/ProgramPlans/ProgramPlanDayModel.ts
import type { ProgramPlanDayStatus } from "../Enums/ProgramPlanDayStatus";
import type { ProgramPlanDayType } from "../Enums/ProgramPlanDayType";

export interface ProgramPlanDayModel
{
	id: number;
	programPlanId: number;
	scheduledDate: string;
	originalScheduledDate?: string;
	dayType: ProgramPlanDayType;
	status: ProgramPlanDayStatus;
	workoutTemplateId?: number;
	workoutTemplateName?: string;
	estimatedDurationMinutes?: number;
	exerciseCount: number;
	startedWorkoutId?: number;
	completedWorkoutId?: number;
	notes?: string;
}

File: server/FitMate.Core/JsonModels/ProgramPlans/ProgramPlanDayModel.cs
using FitMate.DB.Enums;

namespace FitMate.Core.JsonModels.ProgramPlans;

public class ProgramPlanDayModel
{
    public long Id { get; set; }
    public long ProgramPlanId { get; set; }
    public DateOnly ScheduledDate { get; set; }
    public DateOnly? OriginalScheduledDate { get; set; }
    public ProgramPlanDayType DayType { get; set; }
    public ProgramPlanDayStatus Status { get; set; }
    public long? WorkoutTemplateId { get; set; }
    public string? WorkoutTemplateName { get; set; }
    public int? EstimatedDurationMinutes { get; set; }
    public int ExerciseCount { get; set; }
    public long? StartedWorkoutId { get; set; }
    public long? CompletedWorkoutId { get; set; }
    public string? Notes { get; set; }
}
